src/qt/optionsmodel.cpp

```diff
@@ -350,9 +362,13 @@ bool OptionsModel::setData(const QModelIndex & index, const QVariant & value, in
             fMinimizeToTray = value.toBool();
             settings.setValue("fMinimizeToTray", fMinimizeToTray);
             break;
+        case MapPortNatpmp: // core option - can be changed on-the-fly
+            settings.setValue("fUseNatpmp", value.toBool());
+            m_node.mapPort(value.toBool(), MapPort::NAT_PMP);
```

Comment:
This logic won't work if the thread is already running...

Reply:
Going to fix it.

Reply:
Fixed in the latest push.